### PR TITLE
Build: add stylelint pre-commit hook

### PR DESCRIPTION
**Pull request by brendanmckeown**
This PR adds a pre-commit hook (using husky) that runs stylelint on any staged .css files (using lint-staged). Stylelint will automatically fix any issues it finds, and then add updated files to the commit.

### CODE REVIEW

**Comment by tikaro** [comment]
Brendan, I merged @geekychristine 's PR, which means that this now has a conflict. Would you mind using your preferred method to resolve the conflict, then merge your code?

**Review by tikaro — APPROVED** [review]
Once the conflicts are resolved, this is approved!

File: package.json [hunk 1]
```diff
@@ -11,7 +11,24 @@
     "watermark": "./scripts/watermark.sh",
     "dev": "node server.js"
   },
+  "lint-staged": {
+    "*.css": [
+      "stylelint --fix",
```
Comment (tikaro) [inline package.json:16]: I'm curious -- where is the code that limits linting to _only_ files that have been staged? Is it something that husky already knows how to do?

File: package.json [hunk 2]
```diff
@@ -11,7 +11,24 @@
...
+    "hooks": {
+      "pre-commit": "lint-staged"
+    }
+  },
   "dependencies": {
     "express": "^4.17.1"
+  },
+  "devDependencies": {
+    "husky": "^3.1.0",
+    "lint-staged": "^9.5.0",
```
Comment (tikaro) [inline package.json:30]: Oh, I see, this is something lint-staged knows how to do! https://www.npmjs.com/package/lint-staged

**Review by tikaro — APPROVED** [review]
Whoops, I didn't mean to start a review with that last comment. Completing this review with "Approved" to close that open loop.

**Comment by brendanmckeown** [comment]
Forgot to add that when there are unfixable errors in the CSS, the commit will be blocked:
```
▶ git commit -m "test unfixable error in css"
husky > pre-commit (node v12.13.1)
  ↓ Stashing changes... [skipped]
    → No partially staged files found...
  ❯ Running tasks...
    ❯ Running tasks for *.css
      ✖ stylelint --fix
        git add



✖ stylelint --fix found some errors. Please fix them and try committing again.

dist/css/style.css
166:11  ✖  Unexpected unknown unit "a"   unit-no-unknown
husky > pre-commit hook failed (add --no-verify to bypass)
```

**Comment by brendanmckeown** [comment]
Rebased onto master and resolved all conflicts

**Pull request merged**